### PR TITLE
[MM-30932] Adding support for Rotator tool in cluster upgrades.

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Adding support for Rotator tool in cluster upgrades
- This is pending [PR](https://github.com/mattermost/rotator/pull/1)

Once rotator gets merged, the go.mod will be updated here.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-30931
- https://mattermost.atlassian.net/browse/MM-30932
- 
#### Release Note
```release-note
- Adding support for Rotator tool in cluster upgrades
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,18 +15,19 @@ require (
 	github.com/jmoiron/sqlx v1.2.0
 	github.com/lib/pq v1.8.0
 	github.com/mattermost/mattermost-operator v1.13.0
+	github.com/mattermost/rotator v0.0.0-20210225115736-5e1a4be5925b
```
Comment (gabrieljackson) [inline go.mod:18]: Can't wait to get this in! Let's hold off a bit and get the rotator PR merged first so we can point our module at that commit (or tag if we decide that is worth it).

Comment (stylianosrigas) [reply go.mod:18]: Yeap that will be blocked until the rotator gets merged and we release v0.1.0. I just created the PR to track all other changes ;)

File: cmd/cloud/cluster.go [hunk 2]
```diff
@@ -71,6 +71,12 @@ func init() {
 	clusterUpgradeCmd.Flags().String("cluster", "", "The id of the cluster to be upgraded.")
 	clusterUpgradeCmd.Flags().String("version", "", "The Kubernetes version to target. Use 'latest' or versions such as '1.16.10'.")
 	clusterUpgradeCmd.Flags().String("kops-ami", "", "The AMI to use for the cluster hosts. Use 'latest' for the default kops image.")
+	clusterUpgradeCmd.Flags().Bool("use-rotator", true, "Whether the cluster will be upgraded using the node rotator.")
+	clusterUpgradeCmd.Flags().Int("max-scaling", 5, "The maximum number of nodes to rotate every time. If the number is bigger than the number of nodes the number of nodes is the maximum number.")
```
Comment (angeloskyratzakos) [inline cmd/cloud/cluster.go:75]:
```suggestion
	clusterUpgradeCmd.Flags().Int("max-scaling", 5, "The maximum number of nodes to rotate every time. If the number is bigger than the number of nodes, then the number of nodes will be the maximum number.")
```

File: go.mod (hunk 1 above)
Comment (stylianosrigas) [reply go.mod]: This should be ready to review now :)

File: go.mod [hunk 3]
```diff
@@ -15,18 +15,19 @@ require (
...
 	github.com/spf13/cobra v1.1.1
 	github.com/stretchr/testify v1.7.0
 	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/api v0.18.15
+	k8s.io/api v0.19.2
 	k8s.io/apiextensions-apiserver v0.18.8
-	k8s.io/apimachinery v0.18.15
-	k8s.io/client-go v0.18.15
+	k8s.io/apimachinery v0.19.2
+	k8s.io/client-go v0.19.2
```
Comment (gabrieljackson) [inline go.mod:31]: Hmm, this is getting ahead of the operator imports. Do we know if we want to include these changes?

Comment (stylianosrigas) [reply go.mod]: You are right. These changes were brought by the rotator package that was using newer versions. A new rotator release was cut to use the same versions as the provisioner and it is good now.

**Review by gabrieljackson — APPROVED** [review]
Awesome work!

**Review by angeloskyratzakos — APPROVED** [review]
LGTM

**Pull request merged**